src/actions.js

```diff
@@ -77,11 +112,41 @@ export function loadRestaurants() {
 }
 
 export function loadRestaurant({ restaurantId }) {
-  return async (dispatch) => {
-    dispatch(setRestaurant(null));
+  return async (dispatch, getState) => {
+    const id = getState()?.restaurant?.id;
+    const newId = restaurantId;
+
+    if (isNewRestaurant(id, newId)) {
```

Comment:
```suggestion
    const { id } = getState()?.restaurant;

    if (id && isNewRestaurant(id, restaurantId)) {
```

한번 더 추상화 해줄 필요 없이 위와 같이 처리해도 좋겠네요.

그리고 isNewRestaurant가 다른 컴포넌트에서 재사용 되는 경우가 아니라면 굳이 util 쪽에 선언하지 않고 현재 파일의 상단에 선언해줘도 좋겠네요 :)

Reply:
어제 주신 리뷰를 지금 봣네요 ㅠㅠ 그런데 위에 작성해주신 코드가 정상 작동하나요??
초기에 restaurant에는 id라는 키가 없고.. 126번째 줄도 id가 0이거나 없을때 새로운 아이디인지 확인한다고 쓰신거 같은데 무엇인가 이상한 것같습니다..🤔🤔
(id가 0이 아니면 무조건  if가 통과되어서 `dispatch(setRestaurant(null));`이 실행됩니다..ㅠㅠ)

Reply:
아앗 직접 돌려보지 않고 이런식으로 처리하면 좋겠다고 생각해서 남겼던 커맨트인데 올바르게 동작하지 않는군요 🙇🏻‍♂️

id값을 가져와 처리하는 부분은 뭔가 구조분해할당으로 처리하거나 isNewRestaurant 호출이전에 한번 더 체크해서 조금 더 로직을 단순화 할 수 없을까 싶어서 남겼었던 것 같아요ㅎㅎ;

그리고 `const newId = restaurantId;` 이 부분이 한번 더 추상화해주는 느낌이어서 바로 사용하면 좋을 것 같았습니다.

리뷰드릴 코드량이 많아져서 정신없이 리뷰를 남기다보니 오히려 혼란만 드리는 커맨트를 남긴 것 같네요. 주의하겠습니다 🙏